authorize_request.go

```diff
@@ -25,6 +25,11 @@ type AuthorizeRequest struct {
 	State                string    `json:"state" gorethink:"state"`
 	HandledResponseTypes Arguments `json:"handledResponseTypes" gorethink:"handledResponseTypes"`
 
+	// Optional code_challenge as described in rfc7636
+	CodeChallenge string `json:"code_challenge" gorethink:"code_challenge"`
```

Comment:
I really think this can be covered by `Request.Form`

Reply:
I consider that CodeChallenge is like State that's why in coded it here.

Reply:
My first problem was the Session usage, due to the fact it's an interface and shared in openid and oauth2 handler (I fix missing implementation on first commit). I had to store code_challenge to compare it on the token retrieval step, maybe Request.Form is a better place if it is shareable like session.

Reply:
So the `Request.Form` thing will contain all request parameters from the `/auth` endpoint, it will thus also contain `code_challenge` if that parameter is given to the `/auth` endpoint. You won't have to deal with any extra sessions or stuff there.